File: Java_Codes/array_linearsearch.java
import java.util.*;

public class array_linearsearch {
    static int search(int arr[], int n, int key) {
        for (int i = 0; i < n; i++) {
            if (arr[i] == key) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String args[]) {
        int x;
        Scanner scr = new Scanner(System.in);

        System.out.print("Enter the size of the array: ");
        x = scr.nextInt();
        int myArray[] = new int[x];

        System.out.println("Enter the elements of the array: ");
        for (int i = 0; i < x; i++) {
            myArray[i] = scr.nextInt();
        }

        System.out.print("The entered array is: ");
        for (int i = 0; i < myArray.length; i++) {
            System.out.print(myArray[i] + " ");
        }

        System.out.print("\nEnter the value to be searched: ");
        int key = scr.nextInt();

        int position = search(myArray, myArray.length, key);
        if (position == -1) {
            System.out.println("Element is not found. ");
        } else {
            System.out.println("Element is found at index: " + position);
        }
        scr.close();
    }
}

File: C_Codes/array_linearsearch.c
#include<stdio.h>

int linear_search(int arr[], int s, int n){
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == s)
        {
            return i;
            break;
        }
    }
    return -1;
}
int main(){
    int n, s;
    printf("Enter the size of the array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter the elements of the array: ");
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
    }
    
    printf("The entered array is: ");
    for (int i = 0; i < n; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\nEnter the element to be searched in the array: ");
    scanf("%d", &s);

    int result = linear_search(arr, s, n);
    if (result == -1)
    {
        printf("The element is not in the array.");
    }
    else
    {
        printf("The element is at %dth position in the array.", result);
    }
return 0;
}